helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixAdmin.java

```diff
@@ -1035,6 +1035,135 @@ public ClusterManagementMode getClusterManagementMode(String clusterName) {
         : new ClusterManagementMode(status.getManagementMode(), status.getManagementModeStatus());
   }
 
+  @Override
+  public void setPartitionsToError(String clusterName, String instanceName, String resourceName,
+      List<String> partitionNames) {
+    logger.info("Set partitions {} for resource {} on instance {} in cluster {} to ERROR state.",
+        partitionNames == null ? "NULL" : HelixUtil.serializeByComma(partitionNames), resourceName,
+        instanceName, clusterName);
+    sendStateTransitionMessage(clusterName, instanceName, resourceName, partitionNames,
+        Boolean.FALSE);
```

Comment:
reset = false, means error seems a bit wrong.  Can we introduce Enum with 3 values - ERROR, RESET, UNDEFINED 
Pass ERROR and RESET and process it, for undefined, throw exception.

Reply:
makes sense, updated